### DISCUSSION TITLE
Graphical interface for the projector.

### DISCUSSION THREAD

**Discussion by sannelo** [fnuecke/oc2#121] [General]
Is it possible to implement a GUI on a projector?
How to do this, and if it cannot be implemented yet, is it planned in the future?

**Comment by spike11302000** [comment]
it could be theoretically possible to get something like x11 running on it but due to the lack of any mouse input it would be useless. i can try later today to test if it is possible to run x11 on it but still be useless unless mouse input gets added

**Reply by CoolCat467** [reply]
![](https://imgs.xkcd.com/comics/x11.png)

**Reply by spike11302000** [reply]
if i had dollar for every time i had to open xorg.conf i would be the richest man in the world also the most depressed

**Reply by atirutw** [reply]
I doubt OC2 computers have bitmap graphics mode necessary for running GUI programs if that's what you're trying to do.

Some terminals support mouse input, though. Not sure if OC2 does but if it can do that then you could do TUI with mouse input. Should be good enough.

**Reply by spike11302000** [reply]
> I doubt OC2 computers have bitmap graphics mode necessary for running GUI programs if that's what you're trying to do.

The Projector acts as a framebuffer device so you can use the fbdev x11 driver to use the framebuffer to display x11 windows.

> Some terminals support mouse input, though. Not sure if OC2 does

OC2 currently does not support mouse input but i would like it to be a feature at some point

**Reply by atirutw** [reply]
How would you do mouse input for X11 though?

**Reply by spike11302000** [reply]
> How would you do mouse input for X11 though?

Thats the current issue right now as there is no mouse input but x11 also has a feature called "keypad:pointerkeys" which allows you to use your numpad as a mouse input, you can take a look at that [here](https://linuxreviews.org/HOWTO_use_the_numeric_keyboard_keys_as_mouse_in_XOrg)
